x-pack/plugins/fleet/server/routes/package_policy/utils/index.ts

```diff
@@ -53,3 +65,47 @@ export function canUseMultipleAgentPolicies() {
       : 'Reusable integration policies are not supported',
   };
 }
+
+/**
+ * If an agentless agent policy is associated with the package policy,
+ * it will rename the agentless agent policy of a package policy to keep it in sync with the package policy name.
+ */
+export async function renameAgentlessAgentPolicy(
+  soClient: SavedObjectsClientContract,
+  esClient: ElasticsearchClient,
+  packagePolicy: PackagePolicy,
+  name: string
+) {
+  const cloudSetup = appContextService.getCloud();
+  // If cloud and agentless is enabled, we need to rename the agent policy
+  // tech debt: update this condition when Serverless uses the Agentless API
+  // https://github.com/elastic/security-team/issues/9781
+  if (cloudSetup?.isCloudEnabled && appContextService.getExperimentalFeatures().agentless) {
+    const packagePolicyAgentPolicyId = packagePolicy?.policy_id;
+    if (packagePolicyAgentPolicyId) {
+      const agentPolicy = await agentPolicyService.get(soClient, packagePolicyAgentPolicyId);
+
+      const isManagedAgentlessAgentPolicy =
+        agentPolicy?.is_managed && agentPolicy.supports_agentless;
```

Comment:
nit: do we need an additional check for `is_managed` here? Right now afaik all agentless policies are managed and this is an implementation detail, which might change. Tg. if it is decided to have special handling for agentless policies similar to managed but different. Not a big deal, just one potential place to change and it might be confusing for those who are not familiar with the agentless and might wonder "are there non-managed agentless policies?"

Reply:
Good question, the check might not be actually "required" due to creating a policy with the `supports_agentless` flag [currently](https://github.com/elastic/kibana/blob/fdd67f7070037a548494b3830f9cfde64873f1be/x-pack/plugins/fleet/server/services/agent_policy.ts#L340) will add the `is_managed`, but if that eventually changes this check would ensure that only a managed policy gets renamed as it would be an unexpected side effect for a non-managed policy.

Reply:
I am going to remove the is_managed additional checking for now to avoid confusion